pkg/controller/manager/nad/controller.go

```diff
@@ -97,22 +99,23 @@ func (h Handler) OnChange(key string, nad *cniv1.NetworkAttachmentDefinition) (*
 		return nil, err
 	}
 
-	// check annotations
-	if nad.Annotations == nil || nad.Annotations[utils.KeyNetworkConf] == "" {
-		return nad, nil
-	}
-	networkConf, err := utils.NewLayer3NetworkConf(nad.Annotations[utils.KeyNetworkConf])
-	if err != nil {
-		return nil, fmt.Errorf("invalid layer 3 network configure: %w", err)
+	networkConf := &utils.Layer3NetworkConf{}
+	if nad.Annotations != nil && nad.Annotations[utils.KeyNetworkConf] != "" {
+		var err error
+		networkConf, err = utils.NewLayer3NetworkConf(nad.Annotations[utils.KeyNetworkConf])
+		if err != nil {
+			return nil, fmt.Errorf("invalid layer 3 network configure: %w", err)
+		}
 	}
-
 	klog.Infof("netconf: %+v", networkConf)
 
 	if networkConf.CIDR != "" && networkConf.Gateway != "" {
-		// set connectivity as the initial status unknown
+		// initialize connectivity
 		if networkConf.Connectivity == "" {
-			if err := h.setUnknown(nad, networkConf); err != nil {
-				return nil, err
+			if err := h.initializeConnectivity(nad, networkConf); err != nil {
+				klog.Errorf("initialize connectivity of nad %s/%s failed, error: %v", nad.Namespace, nad.Name, err)
```

Comment:
Why don't we return the error here?

Reply:
It makes no severely bad effect if initialize connectivity failed. Moreover, the controller will check and set the connectivity every 15 minutes.